packages/dds/tree/src/forest/editableForest.ts

```diff
@@ -29,6 +30,13 @@ export interface IEditableForest extends IForestSubscription {
      */
     readonly anchors: AnchorSet;
 
+    /**
+     * Create an independent copy of this forest, that uses the provided schema and anchors.
+     *
+     * The new copy will not invalidate observers (dependents) of the old one.
+     */
+    clone(schema: StoredSchemaRepository, anchors: AnchorSet): IEditableForest;
```

Comment:
I think it's may make sense for the `IForestSubscription.clone` to return an `IEditableForest` (which would make this declaration redundant), but maybe there's reasons no to and in any case, that can wait. We can see what Craig thinks when he gets back.
Feel free to resolve this thread without further changes.

Reply:
I chose to do it this way because it seems like a weird circular dependency to have the `IForestSubscription.clone` return an `IEditableForest`. I'm not in any rush to merge this in, so we can just wait for Craig to get back and maybe give him a chance to review if he wants to as well.

Reply:
From a layering perspective, it makes sense to do what you have done here. From an ownership/mutability perspective, having IForestSubscription.clone return an IEditableForest makes more sense (if you have a subscription, and you want to do a transaction, you need to clone, and its safe to do so)

Since IForestSubscription and IEditableForest are both part of the forest module, the layering concern is less important that it would be otherwise. Also since both of those are just types and not runtime objects, the cyclic dep import can't really an issue either (might be nice to use `import type` for it though).

I don't think we will actually want any forest subscriptions that arn't actually editable forests themselves, but even if we had one, it could still support being cloned into one, so I don't see having this method here as a big problem for separation/testability of implementations.

Another thing to consider is documentation: does the cycle make the docs harder to understand. I don't think so: these two types are basically the readonly and mutable views of forest, so having them relate to eachother isn't confusing. If forest subscription was more of a snapshot than a subscription, then I think we would want to keep them separate, but since it has to handle changing over time anyway, I think coupling it to editing is reasonable.

**Thus I think Yann's suggestion is the better approach, though it is subjective as there are good reasons to do either one.** It's also the simplest.

One alternative to both these we could take (which would be the right choice if we were using a snapshot style API instead of a subscription) would be to provide a free function that makes an editable copy of a forest (or an alternate constructor for one type of forest that takes in a forest), and provide an optional acceleration API on forest (maybe via symbol and separate interface, or via optional method) that it uses, if present, to accelerate the cloning (or maybe implement it using the to be implemented fast path underlying data access from forest chunks). I don't think it's worth the complexity to do that here, but its a good design pattern to keep in mind for cloning/factory patterns.